cpp/include/cudf/datetime.hpp

```diff
@@ -190,6 +190,39 @@ std::unique_ptr<cudf::column> add_calendrical_months(
   cudf::column_view const& months,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief  Adds or subtracts a number of months from the date time type and returns a
+ * timestamp column that is of the same type as the input `timestamps` column.
+ *
+ * For a given row, if the `timestamps` value is null, the output for that row is null.
+ * A null months scalar would result in an all null column.
+ * This method preserves the input time and the day where applicable. The date is rounded
+ * down to the last day of the month for that year, if the new day is invalid for that month.
+ * @code{.pseudo}
+ * Example:
+ * timestamps = [5/31/20 08:00:00, 6/30/20 00:00:00, 7/31/20 13:00:00]
+ * months     = -3
+ * output is [2/29/20 08:00:00, 3/30/20 00:00:00, 4/30/20 13:00:00]
+ *
+ * timestamps = [4/28/20 04:00:00, 5/30/20 01:00:00, 6/30/20 21:00:00]
+ * months     = 1
+ * output is [5/28/20 04:00:00, 6/30/20 01:00:00, 7/30/20 21:00:00]
+ *
+ * @endcode
+ *
+ * @param[in] timestamps cudf::column_view of timestamp type.
+ * @param[in] months cudf::scalar of integer type containing the number of months to add.
+ *
+ * @returns cudf::column of timestamp type containing the computed timestamps.
+ * @throw cudf::logic_error if `timestamps` datatype is not a TIMESTAMP or if `months` datatype
+ * is not INT16.
```

Comment:
Why do we have this limitation? Is it to avoid template instantiation explosion?

Reply:
I believe so... This may be relaxed to also support `int32` per #8865

Reply:
According to https://github.com/rapidsai/cudf/issues/8865#issuecomment-896261849 it would be beneficial to also support int32 here.

Reply:
Updated so that this PR supports `int32` for both scalar and column. So it will also close #8865.